pages/search.vue

```diff
@@ -0,0 +1,167 @@
+      <v-tabs v-model="searchTab" centered>
+        <v-tab :key="0">{{ $t('search.topResults') }}</v-tab>
+        <v-tab :key="1">{{ $t('movies') }}</v-tab>
+        <v-tab :key="2">{{ $t('shows') }}</v-tab>
+        <v-tab :key="3">{{ $t('albums') }}</v-tab>
+        <v-tab :key="4">{{ $t('songs') }}</v-tab>
+        <v-tab :key="5">{{ $t('books') }}</v-tab>
+        <v-tab :key="6">{{ $t('people') }}</v-tab>
+        <v-tab :key="7">{{ $t('artists') }}</v-tab>
+      </v-tabs>
+    </v-app-bar>
+    <v-container class="after-second-toolbar">
+      <v-row>
+        <v-col>
+          <v-tabs-items v-model="searchTab" class="transparent">
+            <v-tab-item :key="0">
+              <item-grid :items="topSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="1">
+              <item-grid :items="movieSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="2">
+              <item-grid :items="showSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="3">
+              <item-grid :items="albumSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="4">
+              <item-grid :items="trackSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="5">
+              <item-grid :items="bookSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="6">
+              <item-grid :items="personSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="7">
+              <item-grid :items="artistSearchResults" :loading="loading" />
+            </v-tab-item>
+          </v-tabs-items>
+        </v-col>
+      </v-row>
+    </v-container>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { mapActions } from 'vuex';
+import { BaseItemDto } from '@jellyfin/client-axios';
+
+export default Vue.extend({
+  data() {
+    return {
+      loading: false,
+      searchTab: 0,
+      topSearchResults: [] as BaseItemDto[],
+      movieSearchResults: [] as BaseItemDto[],
+      showSearchResults: [] as BaseItemDto[],
+      albumSearchResults: [] as BaseItemDto[],
+      trackSearchResults: [] as BaseItemDto[],
+      bookSearchResults: [] as BaseItemDto[],
+      personSearchResults: [] as BaseItemDto[],
+      artistSearchResults: [] as BaseItemDto[]
+    };
+  },
+  computed: {
+    searchQuery(): string {
+      return this.$store.state.search.query;
+    }
+  },
+  watch: {
+    searchQuery(newQuery: string, _oldQuery: string): void {
+      if (newQuery === '') {
+        this.$router.back();
+      } else {
+        this.performSearch();
+      }
+    }
+  },
+  beforeMount() {
+    if (this.searchQuery === '') {
+      this.$router.back();
+    }
+  },
+  destroyed() {
+    this.setSearchQuery({ query: '' });
+  },
```

Comment:
I'd avoid a store for this unique goal, and just pass it from `default` as a prop and use an event or something to clear it in the parent

Reply:
We need a store to be able to pass the query from `default` layout to search page

Reply:
Why though, couldn't a prop do it?

Reply:
Because we don't use the search page as a component, rather as a page, so in the code there is never <search-page />, it lives inside <nuxt> inside the default layout

Reply:
Oh yeah that's right, it didn't tick. Couldn't then the query be put into the URL?

Reply:
That's not a bad shout

Reply:
I've updated it to use query params. I haven't squashed it, so once it is all good and approved, then I'll squash...